internal/xds/translator/extension.go

```diff
@@ -53,9 +55,7 @@ func processExtensionPostRouteHook(route *routev3.Route, vHost *routev3.VirtualH
 
 	// If the extension returned a modified Route, then copy its to the one that was passed in as a reference
 	if modifiedRoute != nil {
-		// Overwrite the pointer for the original route.
-		// Uses nolint because of the ineffectual assignment check
-		route = modifiedRoute //nolint
```

Comment:
why not just 

```
*route = *modifiedRoute
```

And 

```
*vHost = *modifiedVH
```

Reply:
This is what I initially wanted to do, but unfortunately it causes a lint error because all the xDS resources have mutexes in the message state fields :sweat_smile:

```
copylocks: assignment copies lock value to *route: github.com/envoyproxy/go-control-plane/envoy/config/route/v3.Route contains google.golang.org/protobuf/internal/impl.MessageState contains sync.Mutex
```